stratum/lib/p4runtime/sdn_controller_manager.cc

```diff
@@ -0,0 +1,445 @@
+// Copyright 2020 Google LLC
+// Copyright 2021-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "stratum/lib/p4runtime/sdn_controller_manager.h"
+
+#include "absl/numeric/int128.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_join.h"
+#include "absl/types/optional.h"
+#include "glog/logging.h"
+#include "p4/v1/p4runtime.pb.h"
+#include "stratum/glue/gtl/map_util.h"
+
+// TODO(max): version 1.3.0 of P4Runtime does not fully support roles yet.
```

Comment:
is this a current TODO or for a future pr ?

Reply:
future, needs a new p4rt release